wordpress-mobile/WordPress-Android
# Issue/sync bp reminders swith with store

Corresponding flux-c PR is [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2418).

This PR allows for saving of the "include blogging prompt" setting in Blogging Reminders. We are also using this value when determining if we should show blogging reminder or blogging prompt.

At the moment we do not use any new reminder related values from the SiteModel, so there is no real way to test it, apart from adding the breakpoint and checking them manually.

To test DB migration:
- Install the version of this app from trunk and login into it.
- Install version from this PR on-top of the version from the trunk, and make sure it launches correctly.

To test:
- Enable Blogging Prompt feature flag and restart the app.
- Navigate to site settings, and setup and blogging reminder that includes a prompt.
- Close the reminder flow, and open it again.
- Confirm that the "Include blogging prompt" switch is ON.
- Advance the time on the device to trigger the Blogging Reminder notification.
- Confirm that the notification includes blogging prompt.
- Turn OFF the "Include blogging prompt" switch in reminder settings.
- Advance the time on the device to trigger the Blogging Reminder notification.
- Confirm that the notification is not blogging prompt notification.


## Regression Notes
1. Potential unintended areas of impact
- DB migration fails.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Test for Rooom migration.

PR submission checklist:
